### PR TITLE
Truncate blog posts

### PR DESCRIPTION
**Pull request by garyanewsome**
###### Truncate Blog Posts

Truncated blog posts at 250 characters. Used sanitize to remove HTML from posts, this removes all of the HTML markup as well as the images.

###### Needs
- code review

### CODE REVIEW

File: app/views/posts/index.haml [hunk 1]
```diff
@@ -6,7 +6,7 @@
           %h2= post.title
           %h6= post.created_at.strftime('%b %d, %Y')
         %div.blog
-          = raw(post.body)
+          = raw truncate(post.body, length: 250)
```
Comment (ninjapanzer) [inline app/views/posts/index.haml]: So you need to strip the HTML tags from the text so there is no way to accidentally split in the middle of an image tag or something.

Your regex must just match all html tags and leave the bodies behind just to calculate the truncation

You will use `gsub` likely and some regex from the internets

**Comment by garyanewsome** [comment]
screen shot 2016-10-21 at 3 05 28 pm
screen shot 2016-10-21 at 3 06 02 pm
screen shot 2016-10-21 at 3 06 27 pm
screen shot 2016-10-21 at 3 06 47 pm

**Comment by garyanewsome** [comment]
Colors will be addressed in next PR

File: app/views/posts/index.haml [hunk 2]
```diff
@@ -6,7 +6,8 @@
           %h2= post.title
           %h6= post.created_at.strftime('%b %d, %Y')
         %div.blog
-          = raw truncate(post.body, length: 250)
+          / = raw sanitize(truncate(post.body, length: 250, separator: ' '))
+          = raw sanitize(post.body, :tags=>[]).truncate(250, :separator => " ").html_safe
```
Comment (ninjapanzer) [inline app/views/posts/index.haml]: Now that you are sanitizing it I don' think you need the raw or html safe anymore

Comment (ninjapanzer) [inline app/views/posts/index.haml]: super cool I didn't know about any of these things

Comment (garyanewsome) [inline app/views/posts/index.haml]: Removed `raw` and `html_safe`

**Comment by ninjapanzer** [comment]
out

Something odd when clicking on the post

**Pull request merged**